File: SuperLauncher/ui/pages/news_page.py
from PyQt6.QtWidgets import QWidget, QVBoxLayout, QLabel, QScrollArea
from PyQt6.QtCore import Qt

class NewsPage(QWidget):
    def __init__(self, main):
        super().__init__()
        layout = QVBoxLayout(self)
        layout.setContentsMargins(15, 15, 15, 15)

        title = QLabel("Новости")
        title.setStyleSheet("font-size: 26px; font-weight: bold; color: white;")
        layout.addWidget(title)

        scroll = QScrollArea()
        scroll.setWidgetResizable(True)
        container = QWidget()
        cl = QVBoxLayout(container)
        cl.setSpacing(12)

        news = [
            ("2025-08-12 v1.4.0.7", "Добавлен Discord RPC"),
            ("2025-07-24 v1.4.0.5", "Поддержка скачивания модов из Modrinth и настройки лаунчера"),
            ("2025-07-23 v1.4.0.4", "Создание и управление Minecraft-серверами"),
            ("2025-07-23 v1.4.0.3", "Новый дизайн и восстановлен код"),
            ("2025-06-26 v1.4.0.2", "Новый дизайн, но утерян код"),
            ("2025-06-26 v1.3", "Лаунчер выйдет из беты"),
        ]

        for date, text in news:
            lbl = QLabel(f"<b>{date}</b>: {text}")
            lbl.setWordWrap(True)
            lbl.setStyleSheet("font-size: 14px; color: #ccc; padding: 10px; background: rgba(255,255,255,0.03); border-radius: 8px;")
            cl.addWidget(lbl)

        cl.addStretch()
        scroll.setWidget(container)
        layout.addWidget(scroll, 1)

File: SuperLauncherCpp/src/ui/pages/NewsPage.cpp
#include "NewsPage.h"
#include "src/network/ApiClient.h"
#include <QHBoxLayout>
#include <QLabel>

NewsPage::NewsPage(ApiClient *client, QWidget *parent)
    : QWidget(parent), m_client(client)
{
    setupUi();
}

void NewsPage::setupUi()
{
    auto *mainLayout = new QVBoxLayout(this);
    mainLayout->setContentsMargins(24, 24, 24, 24);

    auto *title = new QLabel("Новости", this);
    title->setStyleSheet("color: white; font-size: 22px; font-weight: bold;");
    mainLayout->addWidget(title);

    // Controls
    auto *ctrlRow = new QHBoxLayout();
    m_langCombo = new QComboBox(this);
    m_langCombo->addItems({"Русский", "English"});
    m_langCombo->setStyleSheet(
        "QComboBox { background: rgba(255,255,255,0.08); color: white; "
        "border: 1px solid rgba(255,255,255,0.15); border-radius: 6px; "
        "padding: 6px; }");

    m_refreshBtn = new QPushButton("Обновить", this);
    m_refreshBtn->setStyleSheet(
        "QPushButton { background: #7C4DFF; color: white; border: none; "
        "border-radius: 6px; padding: 8px 20px; }"
        "QPushButton:hover { background: #9C6DFF; }");

    connect(m_refreshBtn, &QPushButton::clicked, this, [this]() {
        loadNews(m_langCombo->currentIndex() == 0 ? "ru" : "en");
    });

    ctrlRow->addWidget(m_langCombo);
    ctrlRow->addWidget(m_refreshBtn);
    ctrlRow->addStretch();
    mainLayout->addLayout(ctrlRow);

    // News viewer
    m_newsViewer = new QTextBrowser(this);
    m_newsViewer->setOpenExternalLinks(true);
    m_newsViewer->setStyleSheet(
        "QTextBrowser { background: rgba(255,255,255,0.03); border: 1px solid "
        "rgba(255,255,255,0.08); border-radius: 8px; color: #CCCCCC; "
        "padding: 16px; font-size: 14px; }");
    mainLayout->addWidget(m_newsViewer, 1);
}

void NewsPage::loadNews(const QString &lang)
{
    QString url = "https://raw.githubusercontent.com/anomalyco/SuperLauncher/main/"
                  + lang + "_news.html";
    m_client->getRaw(url, [this](bool ok, const QByteArray &data) {
        if (ok)
            renderNews(QString::fromUtf8(data));
        else
            m_newsViewer->setHtml("<p style='color:#888;'>Не удалось загрузить новости</p>");
    });
}

void NewsPage::renderNews(const QString &html)
{
    m_newsViewer->setHtml(html);
}
